### PR TITLE
Replaces function values with tuples for Guardian :secret_key option

### PR DESCRIPTION
**Pull request by coryodaniel**
``` elixir
def MySecretKey do
  def fetch do
    System.get_env("SECRET_KEY_PASSPHRASE") |> JOSE.JWK.from_file(System.get_env("SECRET_KEY_FILE"))
  end  
end

config :guardian, Guardian,
  allowed_algos: ["Ed25519"],
  secret_key: {MySecretKey, :fetch}
```

Resolves issue #221

### CODE REVIEW

File: lib/guardian.ex [hunk 1]
```diff
@@ -298,8 +298,9 @@ defmodule Guardian do
 
   defp jose_jwk(the_secret = %JOSE.JWK{}), do: the_secret
   defp jose_jwk(the_secret) when is_binary(the_secret), do: JOSE.JWK.from_oct(the_secret)
-  defp jose_jwk(the_secret) when is_function(the_secret, 0), do: the_secret.()
   defp jose_jwk(the_secret) when is_map(the_secret), do: JOSE.JWK.from_map(the_secret)
+  defp jose_jwk({mod, fun}), do: jose_jwk(:erlang.apply(mod, fun))
+  defp jose_jwk({mod, fun, arity}), do: jose_jwk(:erlang.apply(mod, fun, arity))
```
Comment (shantiii) [inline lib/guardian.ex]: The third parameter of the tuple is the argument list, not the arity.

Comment (doomspork) [inline lib/guardian.ex]: @coryodaniel it's nitpicky but could you fix this and rename the parameter?

Comment (coryodaniel) [inline lib/guardian.ex]: Yep!

**Comment by doomspork** [comment]
Thank you @coryodaniel!

**Comment by doomspork** [comment]
@hassox I'm all for this change, any thoughts or concerns before I merge and cut a new version?  This _will_ be a breaking change.

**Comment by hassox** [comment]
Ohh I like it! This is great. Thanks for contributing. Could you please update the Changelog to include this new addition?

**Comment by coryodaniel** [comment]
@hassox Want me to add that to the changelog or was that a note for @doomspork when a new release is made... I wasn't sure where to add it

**Comment by coryodaniel** [comment]
Heyoooo, I didn't cover a test cast that had an issue. Just fixed it. Sorry!

**Comment by doomspork** [comment]
@coryodaniel he was referring to you about the changelog but I can do that if you'd rather 😀

**Comment by coryodaniel** [comment]
@doomspork should I put it under the bug section or add a new release version?

**Comment by doomspork** [comment]
Thank you @coryodaniel 👍

**Comment by Philpax** [comment]
FYI, I just spent some time wracking my head over why the example in the repo's readme (i.e. the tuple format) wouldn't work. As I'm using the latest available public release (i.e. _not_ master), I discovered I had to use the old anonymous function syntax (i.e. `secret_key: fn -> keyGenerationHere end`).

To this end, I suggest adding a note to the readme about this change and removing it when it's time for a new release; might prevent others from getting stuck in the quagmire that I did 😅

**Pull request merged**